### DISCUSSION TITLE
Revisit National Project Concept

### DISCUSSION THREAD

**Discussion by philipbaileynar** [Riverscapes/riverscapes-tools#123] [General]
Riverscapes context keeps accumulating datasets. See @joewheaton requests #122 and #113 

1. The National Project is already gargantuan. Is it time to split it into individual projects by either theme or data source (e.g. all LandFire versions under one roof)?
2. Riverscapes context command line is unsustainable if each dataset must be passed in separately. What is the long term solution?

**Comment by MattReimer** [comment]
> Riverscapes context command line is unsustainable if each dataset must be passed in separately

I'm not too worried about RSContext "Usability" on the command line. That tool will likely only ever be run by us and only now in an automated way.  Let's not agonize too much over making RSContext flexible. It's a totally bespoke ... well "tool" seems generous. It's the product of RSContext that needs to be useable in lots of places, not the python code itself.  

That said, we could always switch to a data-file input like json or yml (or even XML but I can't see a good reason for that) and then you fill in the paths you have and RSContext just works on that.

> Is it time to split it into individual projects by either theme or data source

I'm generally in favour of splitting the National project up. The logical organization we can discuss but as long as we keep things to under a few Gb per project I think that's a good rule of thumb. I'm also fine just splitting out the big ones and leaving a national project behind to collect the smaller ones. Whatever makes the most sense.
